projects/plutus/src/x86/ltc_adc.c

```diff
@@ -0,0 +1,90 @@
+#include "ltc_adc.h"
+
+// x86 module for mocking current sense data
+
+#include "critical_section.h"
+#include "ltc2484.h"
+
+#define TEST_INPUT_VOLTAGE 1000
+
+static void prv_ltc_adc_read(SoftTimerID timer_id, void *context) {
+  LtcAdcStorage *storage = (LtcAdcStorage *)context;
+
+  storage->buffer.status = ltc2484_raw_adc_to_uv(NULL, &storage->buffer.value);
+
+  if (storage->callback != NULL) {
+    storage->callback(&storage->buffer.value, storage->context);
+  }
+
+  soft_timer_start_millis(LTC2484_MAX_CONVERSION_TIME_MS, prv_ltc_adc_read, storage,
+                          &storage->buffer.timer_id);
+}
+
+StatusCode ltc_adc_init(LtcAdcStorage *storage, const LtcAdcSettings *settings) {
+  if (storage == NULL || settings->filter_mode >= NUM_LTC_ADC_FILTER_MODES) {
+    return status_code(STATUS_CODE_INVALID_ARGS);
+  }
+
+  storage->buffer.status = STATUS_CODE_UNINITIALIZED;
+  storage->buffer.value = INT16_MAX;
+  storage->callback = NULL;
+  storage->context = NULL;
+  storage->spi_port = settings->spi_port;
```

Comment:
Why bother copying anything about SPI? The x86 layer doesn't use SPI.

Reply:
Removed